index.js

```diff
@@ -72,47 +23,43 @@ module.exports = (number, options) => {
 
 	options = Object.assign({bits: false, binary: false}, options);
 
-	const UNITS = options.bits ?
-		(options.binary ? BIBIT_UNITS : BIT_UNITS) :
-		(options.binary ? BIBYTE_UNITS : BYTE_UNITS);
+	const UNITS_FIRSTLETTER = (options.bits ? 'b' : 'B') + 'kMGTPEZY';
 
 	if (options.signed && number === 0) {
-		return ` 0 ${UNITS[0]}`;
+		return ` 0 ${UNITS_FIRSTLETTER[0]}`;
 	}
 
-	const isNegative = number < 0;
-	const prefix = isNegative ? '-' : (options.signed ? '+' : '');
-
-	if (isNegative) {
-		number = -number;
-	}
+	const prefix = number < 0 ? '-' : (options.signed ? '+' : '');
+	number = Math.abs(number);
 
-	let localeOptions;
-
-	if (options.minimumFractionDigits !== undefined) {
-		localeOptions = {minimumFractionDigits: options.minimumFractionDigits};
-	}
-
-	if (options.maximumFractionDigits !== undefined) {
-		localeOptions = Object.assign({maximumFractionDigits: options.maximumFractionDigits}, localeOptions);
-	}
+	const localeOptions = Object.fromEntries(
+		Object.entries(options)
+		.filter(([k, _]) => k.includes('FractionDigits'))
+	);
 
 	if (number < 1) {
 		const numberString = toLocaleString(number, options.locale, localeOptions);
-		return prefix + numberString + ' ' + UNITS[0];
+		return prefix + numberString + ' ' + UNITS_FIRSTLETTER[0];
 	}
 
-	const exponent = Math.min(Math.floor(options.binary ? Math.log(number) / Math.log(1024) : Math.log10(number) / 3), UNITS.length - 1);
+	const exponent = Math.min(
+		Math.floor(options.binary ? Math.log(number) / Math.log(1024) : Math.log10(number) / 3),
+		UNITS_FIRSTLETTER.length - 1
+	);
 	// eslint-disable-next-line unicorn/prefer-exponentiation-operator
 	number /= Math.pow(options.binary ? 1024 : 1000, exponent);
 
-	if (!localeOptions) {
+	if (!Object.keys(localeOptions).length) {
```

Comment:
I'm happy to take this and the other object checks if you wanna do a separate pull request though.

Reply:
@sindresorhus, these object controls are tied to lines 35-38 (of my code) which replaced 90-98 (of the project).
Because the project variable `localOptions`, if empty, will be `undefined` on "your" code, but returns an empty object on "mine".
So to change these checks I should also include lines 35-38.

Reply:
Ah, missed that. Never mind then.